cpp/src/barretenberg/honk/sumcheck/sumcheck.test.cpp

```diff
@@ -402,4 +408,304 @@ TEST(Sumcheck, ProverAndVerifierLonger)
+    fr c = a + b;
+    fr d = a + c;
+    uint32_t b_idx = composer.add_variable(b);
+    uint32_t c_idx = composer.add_variable(c);
+    uint32_t d_idx = composer.add_variable(d);
+    for (size_t i = 0; i < 16; i++) {
+        composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+        composer.create_add_gate({ d_idx, c_idx, a_idx, fr::one(), fr::neg_one(), fr::neg_one(), fr::zero() });
+    }
+    // Create a prover (it will compute proving key and witness)
+    auto prover = composer.create_prover();
+
+    // Generate beta and gamma
+    fr beta = fr::random_element();
+    fr gamma = fr::random_element();
+
+    // Compute public input delta
+    const auto public_inputs = composer.circuit_constructor.get_public_inputs();
+    auto public_input_delta =
+        honk::compute_public_input_delta<FF>(public_inputs, beta, gamma, prover.key->circuit_size);
+
+    sumcheck::RelationParameters<FF> relation_parameters{
+        .beta = beta,
+        .gamma = gamma,
+        .public_input_delta = public_input_delta,
+    };
+
+    // Compute grand product polynomial
+    polynomial z_permutation = prover_library::compute_permutation_grand_product<Flavor>(prover.key, beta, gamma);
+
+    ProverPolynomials prover_polynomials;
+
+    prover_polynomials.w_l = prover.key->w_l;
+    prover_polynomials.w_r = prover.key->w_r;
+    prover_polynomials.w_o = prover.key->w_o;
+    prover_polynomials.z_perm = z_permutation;
+    prover_polynomials.z_perm_shift = z_permutation.shifted();
+    prover_polynomials.q_m = prover.key->q_m;
+    prover_polynomials.q_l = prover.key->q_l;
+    prover_polynomials.q_r = prover.key->q_r;
+    prover_polynomials.q_o = prover.key->q_o;
+    prover_polynomials.q_c = prover.key->q_c;
+    prover_polynomials.sigma_1 = prover.key->sigma_1;
+    prover_polynomials.sigma_2 = prover.key->sigma_2;
+    prover_polynomials.sigma_3 = prover.key->sigma_3;
+    prover_polynomials.id_1 = prover.key->id_1;
+    prover_polynomials.id_2 = prover.key->id_2;
+    prover_polynomials.id_3 = prover.key->id_3;
+    prover_polynomials.lagrange_first = prover.key->lagrange_first;
+    prover_polynomials.lagrange_last = prover.key->lagrange_last;
+
+    auto prover_transcript = ProverTranscript<FF>::init_empty();
+
+    auto sumcheck_prover = Sumcheck<Flavor, ProverTranscript<FF>>(prover.key->circuit_size, prover_transcript);
+
+    auto prover_output = sumcheck_prover.execute_prover(prover_polynomials, relation_parameters);
+
+    auto verifier_transcript = VerifierTranscript<FF>::init_empty(prover_transcript);
+
+    auto sumcheck_verifier = Sumcheck<Flavor, VerifierTranscript<FF>>(prover.key->circuit_size, verifier_transcript);
+
+    std::optional verifier_output = sumcheck_verifier.execute_verifier(relation_parameters);
+
+    ASSERT_TRUE(verifier_output.has_value());
+}
+
+/**
+ * @brief Test the Ultra Sumcheck Prover and Verifier for a real circuit
+ *
+ */
+TEST(Sumcheck, RealCircuitUltra)
```

Comment:
Shouldn't there be one (or multiple) tests, or maybe just extending this one, that check ROM Consistency, RAM Timestamp, non-native arithmetic? I know their correctness is checked in the relation folder  but I'm just wary of very subtle bugs.